### PR TITLE
[#375] Add `duplicate` keyword to segment's configuration

### PR DESCRIPTION
**Pull request by FlorentFlament**
This is an experimental feature to allow sharing a segment, by duplicating it into several memory sections.

It allows writing things like that:

    SEGMENTS {
        ZEROPAGE: load = RAM,  type = zp;
        BSS:      load = RAM,  type = bss, optional = yes;
        FXZP:     load = RAM,  type = zp,  optional = yes;

        # Shared segments
        COMMON:   load = ROM0, type = ro,                 duplicate = ROM1, duplicate = ROM2, duplicate = ROM3, duplicate = ROM4;
        CODE:     load = ROM0, type = ro,                 duplicate = ROM1;
        RODATA:   load = ROM0, type = ro, optional = yes, duplicate = ROM1;

        # Initialized data to be copied into RAM will be stored in bank0
        DATA:     load = ROM0, type = rw, optional = yes, define = yes, run = RAM;

        # Per bank data segments
        CODE0:    load = ROM0, type = ro, optional = yes;
        RODATA0:  load = ROM0, type = ro, optional = yes;
        CODE1:    load = ROM1, type = ro, optional = yes;
        RODATA1:  load = ROM1, type = ro, optional = yes;
        CODE2:    load = ROM2, type = ro, optional = yes;
        RODATA2:  load = ROM2, type = ro, optional = yes;
        CODE3:    load = ROM3, type = ro, optional = yes;
        RODATA3:  load = ROM3, type = ro, optional = yes;
        ...
    }

### CODE REVIEW

**Comment by oliverschmidt** [comment]
I must admit that I don't like the "duplicate" approach. I'd like to discuss this rather at https://github.com/cc65/cc65/issues/375 ...

**Comment by FlorentFlament** [comment]
Fair enough, I must admit that I am not fully convinced neither by the "duplicate" approach. Let's discuss this at #375

File: src/ld65/config.c [hunk 1]
```diff
@@ -222,7 +222,7 @@ static MemoryArea* CfgFindMemory (unsigned Name)
 
 
 static MemoryArea* CfgGetMemory (unsigned Name)
-/* Find the memory are with the given name. Print an error on an invalid name */
```
Comment (polluks) [inline src/ld65/config.c]: area?

Comment (FlorentFlament) [inline src/ld65/config.c]: Oh, nice catch

**Comment by FlorentFlament** [comment]
Abandonning the PR for now. I'll try to work on another implementation of #375, as discussed in the issue.

**Pull request closed without merging**